### PR TITLE
Print diagnostic info according to TAP spec; maintenance

### LINKED ISSUE #3: Substitute character
According to docs for TAP you should use `#` for comments not `*` in this https://github.com/vadimdemedes/supertap/blob/master/index.js#L40

https://testanything.org/tap-version-13-specification.html

this is provoking an error with my package (https://github.com/yovasx2/ava-tap-json) when I use new AVa version 2.3.0

### PR DESCRIPTION
**Pull request by tymfear**
TAP specification says that diagnostic info lines should start with `#`.
Fixes #3 

Also:

* Require Node.js 10
* Update dependencies

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,14 +24,14 @@
...
-    "strip-ansi": "^4.0.0"
+    "arrify": "^2.0.1",
+    "indent-string": "^4.0.0",
+    "js-yaml": "^3.14.0",
+    "serialize-error": "^7.0.1",
+    "strip-ansi": "^6.0.0"
   },
   "devDependencies": {
-    "ava": "^0.20.0",
+    "ava": "3.13.0",
```
Comment (novemberborn) [inline package.json]:
```suggestion
    "ava": "^3.13.0",
```

File: test/test.js [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
-		.catch(err => err)
-		.then(result => result.stdout);
+const fs = require('fs');
+const execa = require('execa');
+const test = require('ava');
+const path = require('path');
+
+const exec = async cmd => {
+	const {stdout} = await execa.shell(cmd)
+		.catch(error => error);
```
Comment (novemberborn) [inline test/test.js]: IIRC there's an `execa` option for this.

File: index.js [hunk 3]
```diff
@@ -73,6 +73,6 @@ exports.finish = stats => {
 		`# tests ${passed + failed + skipped}`,
 		`# pass ${passed}`,
 		skipped > 0 ? `# skip ${skipped}` : null,
-		`# fail ${failed + crashed + todo}\n`
-	].filter(Boolean).join('\n');
+		`# fail ${failed + crashed + todo}${os.EOL}`
+	].filter(Boolean).join(os.EOL);
```
Comment (novemberborn) [inline index.js]: What does the TAP spec say about line endings? Any chance it mandates `\n`?

File: .gitignore [hunk 4]
```diff
@@ -1,2 +1,3 @@
 node_modules
 yarn.lock
+.idea
```
Comment (novemberborn) [inline .gitignore]: Ideally this is in your global `.gitignore`.

**Review by novemberborn — COMMENTED** [review]
Thanks for opening the PR @tymfear.

Given the syntax changes you should also update the Node.js engines this is tested with. It should be 10, 12, 14 and 15 I think.

File: test/test.js (hunk 2 above)
Comment (tymfear) [reply test/test.js]: I can't find it, tbh. I haven't worked with `execa`  before, so maybe there is some option for that, but couldn't find it in its README. So would appreciate an advice on what to look at.

File: index.js (hunk 3 above)
Comment (tymfear) [reply index.js]: Haven't seen anything related in docs.

File: .gitignore (hunk 4 above)
Comment (tymfear) [reply .gitignore]: Reverted

File: package.json (hunk 1 above)
Comment (tymfear) [reply package.json]: Done

**Comment by tymfear** [comment]
ping @novemberborn

File: test/test.js (hunk 2 above)
Comment (novemberborn) [reply test/test.js]: `reject: false` but the `execa` version that was used here was quite old.

**Comment by novemberborn** [comment]
Travis is taking too long to run, so I've added GitHub Actions instead. Which isn't running, perhaps because this isn't in master yet. Will merge and fix what needs fixing.

**Pull request merged**